internal/app/singularity/cache_clean_linux.go

```diff
@@ -0,0 +1,201 @@
+			}
+		}
+	}
+
+	return foundMatch, nil
+}
+
+func cleanOciCache(cacheName string) (bool, error) {
+	foundMatch := false
+	blobs, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		return false, fmt.Errorf("Unable to opening cache folder: %v", err)
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
+		if err != nil {
+			return false, fmt.Errorf("Unable to look in cache folder: %v", err)
+		}
+		for _, b := range blob {
+			if b.Name() == cacheName {
+				sylog.Debugf("Removing: %v", filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+				err = os.RemoveAll(filepath.Join(cache.OciTemp(), f.Name()))
+				if err != nil {
+					return false, fmt.Errorf("Unable to remove cache: %v", err)
+				}
+				foundMatch = true
+			}
+		}
+	}
+
+	return foundMatch, nil
+}
+
+// CleanCacheName : will clean a container with the same name as cacheName (in the cache directory).
+// if libraryCache == true; only search thrught library cache. if ociCache == true; only search the
+// oci-tmp cache. if both are false; search all cache, and if both are true; again, search all cache.
+func CleanCacheName(cacheName string, libraryCache, ociCache bool) (bool, error) {
+	if libraryCache == ociCache {
+		matchLibrary, err := cleanLibraryCache(cacheName)
+		if err != nil {
+			return false, err
+		}
+		matchOci, err := cleanOciCache(cacheName)
+		if err != nil {
+			return false, err
+		}
+		if matchLibrary == true || matchOci == true {
+			return true, nil
+		}
+		return false, nil
+	}
+
+	match := false
+	if libraryCache == true {
+		match, err := cleanLibraryCache(cacheName)
+		if err != nil {
+			return false, err
+		}
+		return match, nil
+	} else if ociCache == true {
+		match, err := cleanOciCache(cacheName)
+		if err != nil {
+			return false, err
+		}
+		return match, nil
+	}
+	return match, nil
+}
+
+// CleanSingularityCache : the main function that drives all these other functions, if allClean == true; clean
+// all cache. if typeNameClean contains somthing; only clean that type. if cacheName contains somthing; clean only
+// cache with that name.
+func CleanSingularityCache(cleanAll bool, cacheCleanTypes []string, cacheName string) error {
+	libraryClean := false
+	ociClean := false
+	blobClean := false
+
+	for _, t := range cacheCleanTypes {
+		switch t {
+		case "library":
+			libraryClean = true
+		case "oci":
+			ociClean = true
+		case "blob", "blobs":
+			blobClean = true
+		case "all":
+			cleanAll = true
+		default:
+			sylog.Fatalf("Not a valid type: %v", t)
+			os.Exit(2)
+		}
+	}
+
+	if len(cacheName) >= 1 && cleanAll != true {
+		foundMatch, err := CleanCacheName(cacheName, libraryClean, ociClean)
+		if err != nil {
+			return err
+		}
+		if foundMatch != true {
+			sylog.Warningf("No cache found with givin name: %v", cacheName)
+			os.Exit(0)
+		}
+		return nil
+	}
+
+	if cleanAll == true {
+		if err := cache.Clean(); err != nil {
+			return err
+		}
+	}
+	if libraryClean {
```

Comment:
@WestleyK If you create a general function "CleanCache" sending the type of the cache there, you would just need to verify which type of cache to delete inside that function and here the call would be just something like:

 ``` 
if err := CleanCache(cacheType); err != nil {
			return err
		}
```